### DISCUSSION TITLE
AG Grid call_api_method and multi-line functions

### DISCUSSION THREAD

**Discussion by snowbollaanm** [zauberzeug/nicegui#2152] [Q&A]
### Question

Hi.

I've been tinkering a lot recently with AG Grid trying to bring out its full functionality and there's a couple things I am confused on:

1. I've been trying to use the call_api_method to pin two rows to the bottom and have yet to succeed. I've tried doing this 2 ways with neither working so far: 
```py
circuit_grid.call_api_method('gridApi.setGridOption("pinnedBottomRowData", rows)', *bottomRows), 
```
and
```py
circuit_grid.call_api_method('pinnedBottomRowData', *bottomRows)
```
2. I've been struggling with getting functions to parse through into AG Grid, I have had no problem with small one-line functions like you have on your [demos](https://nicegui.io/documentation/aggrid#ag_grid_with_dynamic_row_height) but passing a function that requires multiple lines such as: 
```py
    getRowStyle: params => {
        if (params.node.rowIndex % 2 === 0) {
            return { background: 'red' };
        }
```
has been very difficult 

Any help on this would be greatly appreciated!

**Comment by falkoschindler** [comment]
I think you don't need to call an API method for setting pinned rows:

```py
ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
    ],
    'rowData': [
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
        {'name': 'Dave', 'age': 30},
        {'name': 'Eve', 'age': 20},
    ],
    'pinnedTopRowData': [
        {'name': 'Alice', 'age': 18},
    ],
    'pinnedBottomRowData': [
        {'name': 'Frank', 'age': 20},
    ],
})
```

And multi-line functions should work like this:

```py
ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 20},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
        {'name': 'Dave', 'age': 30},
        {'name': 'Eve', 'age': 20},
    ],
    ':getRowStyle': '''params => {
            if (params.node.rowIndex % 2 === 0) {
                return { background: 'red' };
            }
        }
    '''
})
```

Note that the leading colon ":" indicates a function, which needs to get evaluated on the client. Triple quotes are used for multi-line strings. Maybe you missed a closing bracket?
